src/engine/renderer/tr_main.cpp

```diff
@@ -2354,39 +2354,48 @@ void R_AddLightInteractions()
 	link_t       *l;
 
 	tr.refdef.numShaderLights = 0;
+
 	for ( i = 0; i < tr.refdef.numLights; i++ )
 	{
 		light = tr.currentLight = &tr.refdef.lights[ i ];
 
-		if ( light->isStatic ) {
-			if ( r_staticLight->integer != 1 || ( ( r_precomputedLighting->integer || tr.worldLight != lightMode_t::MAP ) && !light->noRadiosity ) )
+		if ( light->isStatic )
+		{
+			if ( !glConfig2.staticLight )
 			{
-				if( r_staticLight->integer == 2 ) {
-					tr.refdef.numShaderLights++;
-					tr.pc.c_slights++;
-				}
-
-				light->cull = cullResult_t::CULL_OUT;
 				continue;
 			}
-		} else if ( light->l.inverseShadows ) {
-			if( glConfig2.dynamicLight == 0 ) {
+			else if ( ( r_precomputedLighting->integer || tr.worldLight != lightMode_t::MAP ) && !light->noRadiosity )
+			{
```

Comment:
Here is the code skipping the render of static lights as dynamic ones when `r_precomputedLighting` is enabled.